### PR TITLE
support casting Decimal to String

### PR DESCRIPTION
**Pull request by sperlingxx**
Signed-off-by: sperlingxx <lovedreamf@gmail.com>
Current PR is to support casting Decimal to String, which is requested in issue #1625.

### CODE REVIEW

**Comment by sperlingxx** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala [hunk 1]
```diff
@@ -419,6 +419,9 @@ case class GpuCast(
       case (from: DecimalType, to: DecimalType) =>
         castDecimalToDecimal(input.getBase, from, to)
 
+      case (_: DecimalType, StringType) =>
```
Comment (revans2) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala:422]: I think we have to update the `CastChecks` to allow casting from decimal to string for this to actually impact user queries.

Comment (sperlingxx) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala:422]: Sorry. I forgot it.

Comment (sperlingxx) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala:422]: Fixed.

**Comment by sperlingxx** [comment]
build

**Comment by sperlingxx** [comment]
build

**Comment by sperlingxx** [comment]
build

File: tests/src/test/scala/com/nvidia/spark/rapids/GpuExpressionTestSuite.scala [hunk 2]
```diff
@@ -172,6 +172,11 @@ abstract class GpuExpressionTestSuite extends SparkQueryCompareTestSuite {
     }
   }
 
+  def compareStringifiedDecimals(expected: String, actual: String): Boolean = {
+    (expected == null && actual == null) ||
+        (expected != null && actual != null && Decimal(expected) == Decimal(actual))
```
Comment (revans2) [inline tests/src/test/scala/com/nvidia/spark/rapids/GpuExpressionTestSuite.scala:177]: I thought we wanted to produce the exact same result.  If we don't produce the exact same result then at a minimum we need to document the incompatibility and if it is a big enough difference, then we need to have a config to disable it by default.

**Comment by sperlingxx** [comment]
build

**Comment by sperlingxx** [comment]
build

**Comment by sperlingxx** [comment]
build

**Comment by sperlingxx** [comment]
build

**Comment by sperlingxx** [comment]
I think this PR is ready to merge.

**Review by jlowe — APPROVED** [review]

**Pull request merged**